src/db/mysql/models.rs

```diff
@@ -49,19 +50,27 @@ pub fn run_embedded_migrations(settings: &Settings) -> Result<()> {
     Ok(embedded_migrations::run(&conn)?)
 }
 
+enum CollectionLock {
+    Read,
+    Write,
+}
+
 /// Per session Db metadata
 pub(super) struct MysqlDbSession {
     /// The "current time" on the server used for this session's operations
     pub(super) timestamp: i64,
     /// Cache of collection modified timestamps per (user_id, collection_id)
     coll_modified_cache: HashMap<(u32, i32), i64>,
+    /// Currently locked collections
+    coll_locks: HashMap<(u32, i32), CollectionLock>,
```

Comment:
I'm not totally convinced we need this lock metadata as long as we're careful enough about handling the transactions/locking -- which we should be if it's to be managed by actix_web middleware/extractors (like how the python's handled by decorators).

but I've ported it over anyway

Reply:
This probably makes sense in the prior model where only a single node can access a db at once, and is fine for the MySQL case at least.